src/Adapter/Product/QueryHandler/GetProductCombinationsForEditingHandler.php

```diff
@@ -0,0 +1,138 @@
+use PrestaShop\PrestaShop\Adapter\Product\Repository\CombinationRepository;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\Query\GetProductCombinationsForEditing;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\QueryHandler\GetProductCombinationsForEditingHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\QueryResult\CombinationAttributeInformation;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\QueryResult\CombinationForEditing;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\QueryResult\CombinationListForEditing;
+
+/**
+ * Handles @see GetProductCombinationsForEditing using legacy object model
+ */
+final class GetProductCombinationsForEditingHandler extends AbstractProductHandler implements GetProductCombinationsForEditingHandlerInterface
+{
+    /**
+     * @var CombinationRepository
+     */
+    private $combinationRepository;
+
+    /**
+     * @param CombinationRepository $combinationRepository
+     */
+    public function __construct(
+        CombinationRepository $combinationRepository
+    ) {
+        $this->combinationRepository = $combinationRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(GetProductCombinationsForEditing $query): CombinationListForEditing
+    {
+        $product = $this->getProduct($query->getProductId());
+        $productId = (int) $product->id;
+        $combinations = $this->combinationRepository->getProductCombinations($productId, $query->getLimit(), $query->getOffset());
+
+        $combinationIds = array_map(function ($combination): int {
+            return (int) $combination['id_product_attribute'];
+        }, $combinations);
+
+        $attributesInformation = $this->combinationRepository->getAttributesInfoByCombinationIds(
+            $combinationIds,
+            $query->getLanguageId()
+        );
+
+        return $this->formatCombinationsForEditing(
+            $combinations,
+            $attributesInformation,
+            $this->combinationRepository->getTotalCombinationsCount($productId)
+        );
+    }
+
+    /**
+     * @param array $combinations
+     * @param array<int, array<int, mixed>> $attributesInformationByCombinationId
+     * @param int $totalCombinationsCount
+     *
+     * @return CombinationListForEditing
+     */
+    private function formatCombinationsForEditing(
+        array $combinations,
+        array $attributesInformationByCombinationId,
+        int $totalCombinationsCount
+    ): CombinationListForEditing {
+        $combinationsForEditing = [];
+
+        foreach ($combinations as $combination) {
+            $combinationId = (int) $combination['id_product_attribute'];
+            $combinationAttributesInformation = [];
+
+            foreach ($attributesInformationByCombinationId[$combinationId] as $attributesInfo) {
+                $combinationAttributesInformation[] = new CombinationAttributeInformation(
```

Comment:
We already discussed about how this duplicates the `CombinationAttributeInformation` but it can be optimized later ^^

Reply:
I don't recall what are you talking about, sorry (this PR is ~3 months old already and we've talked alot) :sweat_smile: 
But we can come back to it later if you say its not so important now